### PR TITLE
domain tests v2

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: module/account/tests/Domain/Provider/PrivilegeCodeProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ */
+class PrivilegeCodeProviderTest extends TestCase
+{
+    /**
+     */
+    public function testProviderCreation(): void
```
Comment (rprzedzik) [inline module/account/tests/Domain/Provider/PrivilegeCodeProviderTest.php]: This is not creation test,  this is providing method test

File: module/account/tests/Domain/Provider/PrivilegeDictionaryProviderTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+use PHPUnit\Framework\TestCase;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ */
+class PrivilegeDictionaryProviderTest extends TestCase
+{
+    /**
+     */
+    public function testProvideCreation(): void
```
Comment (rprzedzik) [inline module/account/tests/Domain/Provider/PrivilegeDictionaryProviderTest.php]: This is not creation test, this is providing method test

File: module/account/tests/Domain/Provider/PrivilegeGroupedByAreaProviderTest.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+use Ergonode\Account\Domain\Provider\PrivilegeGroupedByAreaProvider;
+use Ergonode\Account\Domain\Query\PrivilegeQueryInterface;
+use Ergonode\Account\Infrastructure\Resolver\PrivilegeTypeResolverInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ */
+class PrivilegeGroupedByAreaProviderTest extends TestCase
```
Comment (rprzedzik) [inline module/account/tests/Domain/Provider/PrivilegeGroupedByAreaProviderTest.php]: This is not creation test,  this is providing method test

File: module/attribute/tests/Domain/Provider/AttributeParametersProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ */
+class AttributeParametersProviderTest extends TestCase
+{
+    /**
+     */
+    public function testProviderCreation(): void
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/Provider/AttributeParametersProviderTest.php]: This is not creation test,  this is providing method test

File: module/attribute/tests/Domain/Provider/Dictionary/AttributeGroupDictionaryProviderTest.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ */
+class AttributeGroupDictionaryProviderTest extends TestCase
+{
+    /**
+     */
+    public function testProviderCreation(): void
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/Provider/Dictionary/AttributeGroupDictionaryProviderTest.php]: This is not creation test,  this is getDictionary method test

File: module/attribute/tests/Domain/Provider/Dictionary/AttributeTypeDictionaryProviderTest.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+use PHPUnit\Framework\TestCase;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ */
+class AttributeTypeDictionaryProviderTest extends TestCase
+{
+    /**
+     */
+    public function testProviderCreation(): void
```
Comment (rprzedzik) [inline module/attribute/tests/Domain/Provider/Dictionary/AttributeTypeDictionaryProviderTest.php]: This is not creation test,  this is dicionary method test

File: module/category-tree/tests/Domain/Entity/CategoryTreeTest.php [hunk 7]
```diff
@@ -67,4 +68,43 @@ public function testAddingCategory(): void
...
+    {
+        $categoryId = CategoryId::generate();
+        $node = new Node($categoryId);
+        $categories = [$node];
+
+        $tree = new CategoryTree($this->id, $this->code, $this->name);
+
+        $tree->updateCategories($categories);
+
+        $this->assertTrue($tree->hasCategory($categoryId));
```
Comment (rprzedzik) [inline module/category-tree/tests/Domain/Entity/CategoryTreeTest.php]: Add test that check is categorises correct after update, this one only check are any categories, but not correct corrct

Comment (wiewiurdp) [reply module/category-tree/tests/Domain/Entity/CategoryTreeTest.php]: I am adding particular cattegory with Category Id and then checking if this particular category exist in CategoryTree by puting CategoryId (->hasCategory(CategoryId)

**Pull request merged**